ipatests/pytest_plugins/integration/__init__.py

```diff
@@ -55,13 +64,43 @@ def collect_test_logs(node, logs_dict, test_config):
     :param test_config: Pytest configuration
     """
     collect_logs(
-        name=node.nodeid.replace('/', '-').replace('::', '-'),
+        name=_get_logname_from_node(node),
         logs_dict=logs_dict,
         logfile_dir=test_config.getoption('logfile_dir'),
         beakerlib_plugin=test_config.pluginmanager.getplugin('BeakerLibPlugin'),
     )
 
 
+def collect_systemd_journal(node, hosts, test_config):
+    """Collect systemd journal from remote hosts
+
+    :param node: The pytest collection node (request.node)
+    :param hosts: List of hosts from which to collect journal
+    :param test_config: Pytest configuration
+    """
+    name = _get_logname_from_node(node)
+    logfile_dir = test_config.getoption('logfile_dir')
+
+    for host in hosts:
+        log.info("Collecting journal from: %s", host.hostname)
+
+        topdirname = os.path.join(logfile_dir, name, host.hostname)
+        if not os.path.exists(topdirname):
+            os.makedirs(topdirname)
+
+        # Get journal content
+        cmd = host.run_command(
+            ['journalctl', '--since', '-1h'],
+            log_stdout=False, raiseonerr=False)
+        if cmd.returncode:
+            log.error('An error occurred while collecting journal')
+            continue
+
+        # Write journal to file
+        with open(os.path.join(topdirname, "journal"), 'w') as f:
```

Comment:
I expected that you download only chosen services not everything. In this case it can be in /var/log/messages but basically I doesn't care if `messages` or `journal`.